File: bench/primes.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
static long count_primes(long n){
    char *f = (char*)malloc((size_t)n); memset(f,1,(size_t)n);
    long count=0;
    for(long p=2;p<n;p++) if(f[p]){ count++; for(long k=p*p;k<n;k+=p) f[k]=0; }
    free(f); return count;
}
int main(void){ printf("%ld\n", count_primes(2000000)); return 0; }

File: bench/primes.py
def count_primes(n):
    f = bytearray(b'\x01')*n
    count = 0
    for p in range(2, n):
        if f[p]:
            count += 1
            f[p*p:n:p] = b'\x00'*(((n-1-p*p)//p)+1)
    return count
print(count_primes(2000000))
